### ISSUE TITLE
`$ExpectType` did not unwrap the type alias

### ISSUE THREAD

**Issue by ikatyang** [Microsoft/dtslint#57]
This is just a minimal repro, the real use case is for [types/npm-ramda](https://github.com/types/npm-ramda) to generate DT compatible tests so that we could put them on DT, but we [got problems](https://github.com/types/npm-ramda/issues/191#issuecomment-323011442).

Is there any reason we can't pass the `enclosingDeclaration` parameter to `typeToString` in the [expect rule](https://github.com/Microsoft/dtslint/blob/5258797c5219c54d2268f2fdf469a74593598b53/src/rules/expectRule.ts#L299)?

**Code**

(index.d.ts)

```ts
export type Or<T, U> = T | U;
```

(something-tests.ts)

```ts
import * as something from 'something';

declare const result: something.Or<string, number>;

// $ExpectType string | number
result;
```

**Expected**

no errors

**Actual**

```text
ERROR: 6:1   expect   Expected type to be:
  string | number
got:
  Or<string, number>
```

**Comment by danvk** [comment]
Or even more ridiculous:

```ts
ERROR: 299:1  expect                      Expected type to be:
  'Point'
got:
  "Point"
```

The solution being to change single quotes to double quotes.

The root issue here is that dtslint just does a comparison on the string representation of the types:https://github.com/Microsoft/dtslint/blob/4539c93657ea93e96eff84aa835422f6eed3e3b7/src/rules/expectRule.ts#L297-L304

The solution is to change the check to use the [type relationships API](https://github.com/Microsoft/TypeScript/pull/9943). It should at least be `isIdenticalTo` rather than a string comparison.
